### PR TITLE
Remove remote_group_id usage at loadbalancer SGs

### PR DESCRIPTION
**Pull request by luis5tb**
This patch removes the usage of remote_group_id at the loadbalancer
SG rules to allow the access to the namespace. It obtaines the
namespace associated range and use remote_ip_prefix instead.

Note remote_group_id is still used for the default and namespace
SG groups. We can improve it by replacing the rules inside those two
(which use remote_group_ids too) to use remote_ip_prefixes.

### CODE REVIEW

File: kuryr_kubernetes/controller/drivers/lbaasv2.py [hunk 1]
```diff
@@ -403,6 +409,22 @@ def _extend_lb_security_group_rules(self, loadbalancer, listener):
...
+            security_group_id=sg,
+            direction='ingress',
+            description='Kuryr Namespace SG rule')
+
+        if len(sg_rules['security_group_rules']) != 1:
+            LOG.warning("There are more than one Kuryr SG rules associated "
+                        "to the namespace. This means manual addition and "
+                        "may break the isolation and lead to side effects.")
+        # Only 1 should exits. Returning the first one
+        if sg_rules['security_group_rules'][0].get('remote_ip_prefix'):
```
Comment (MaysaMacedo) [inline kuryr_kubernetes/controller/drivers/lbaasv2.py:424]: Why not just` return sg_rules['security_group_rules'][0].get('remote_ip_prefix')`?

Comment (luis5tb) [reply kuryr_kubernetes/controller/drivers/lbaasv2.py:424]: right!

File: kuryr_kubernetes/controller/drivers/lbaasv2.py [hunk 2]
```diff
@@ -349,17 +349,23 @@ def _extend_lb_security_group_rules(self, loadbalancer, listener):
...
-                            'port_range_max': listener.port,
-                            'protocol': listener.protocol,
-                            'security_group_id': sg_id,
-                            'remote_group_id': sg,
-                            'description': listener.name,
-                        },
-                    })
+                    remote_ip_prefix = self._get_remote_ip_from_sg(sg)
+                except n_exc.NeutronClientException as ex:
+                    LOG.exception('Failed when creating security group '
```
[the remark this answers is not in the document]
Comment (luis5tb) [reply kuryr_kubernetes/controller/drivers/lbaasv2.py:354]: done

File: kuryr_kubernetes/controller/drivers/lbaasv2.py [hunk 3]
```diff
@@ -403,6 +409,22 @@ def _extend_lb_security_group_rules(self, loadbalancer, listener):
                               'to enable routes for listener %s.',
                               listener.name)
 
+    def _get_remote_ip_from_sg(sg):
+        neutron = clients.get_neutron_client()
+        sg_rules = neutron.list_security_group_rules(
+            security_group_id=sg,
+            direction='ingress',
+            description='Kuryr Namespace SG rule')
```
[the remark this answers is not in the document]
Comment (luis5tb) [reply kuryr_kubernetes/controller/drivers/lbaasv2.py:417]: ohh, I forgot to commit that... yes!

File: kuryr_kubernetes/controller/drivers/lbaasv2.py [hunk 4]
```diff
@@ -403,6 +410,22 @@ def _extend_lb_security_group_rules(self, loadbalancer, listener):
...
                               listener.name)
 
+    def _get_remote_ip_prefix_from_sg(self, sg):
+        neutron = clients.get_neutron_client()
+        sg_rules = neutron.list_security_group_rules(
+            security_group_id=sg,
+            direction='ingress',
+            description='Kuryr Namespace SG rule')
+
+        if len(sg_rules['security_group_rules']) != 1:
```
Comment (MaysaMacedo) [inline kuryr_kubernetes/controller/drivers/lbaasv2.py:420]: Shouldn't this be > 1?

File: kuryr_kubernetes/controller/drivers/lbaasv2.py [hunk 5]
```diff
@@ -403,6 +447,45 @@ def _extend_lb_security_group_rules(self, loadbalancer, listener):
...
+        if len(sg_rules['security_group_rules']) > 1:
+            LOG.warning("There are more than one Kuryr SG rules associated "
+                        "to the namespace. This means manual addition and "
+                        "may break the isolation and lead to side effects.")
+        # Only 1 should exits. Returning the first one
+        for sg_rule in sg_rules['security_group_rules']:
+            return sg_rule.get('remote_ip_prefix')
+        return ""
+
+    def _get_pool_cidr(self, pool_id):
```
Comment (dulek) [inline kuryr_kubernetes/controller/drivers/lbaasv2.py:466]: This returns list, should be `_get_pool_cidrs`.

File: kuryr_kubernetes/controller/drivers/namespace_security_groups.py [hunk 6]
```diff
@@ -105,18 +106,50 @@ def create_namespace_sg(self, namespace, project_id, crd_spec):
...
+                neutron.create_security_group_rule(
+                    {
+                        "security_group_rule": {
+                            "direction": "ingress",
+                            "remote_ip_prefix": crd_spec['subnetCIDR'],
+                            "security_group_id": (
+                                cfg.CONF.namespace_sg.sg_allow_from_default)
+                        }
+                    })
+                neutron.create_security_group_rule(
```
Comment (dulek) [inline kuryr_kubernetes/controller/drivers/namespace_security_groups.py:142]: The first rule you add doesn't mention any protocol, so most likely will open it for any. Why is the udp one needed?

Comment (luis5tb) [reply kuryr_kubernetes/controller/drivers/namespace_security_groups.py:142]: right... I think while testing it was only creating the tcp one... so I added the udp one... but now it seems it is only adding the rule without protocol... so. I'll double check

**Pull request merged**